src/Form/Type/VichImageType.php

```diff
@@ -71,7 +71,7 @@ public function buildView(FormView $view, FormInterface $form, array $options):
         $view->vars['image_uri'] = null;
         $view->vars['download_uri'] = null;
 
-        if (null !== $object) {
+        if ((null !== $object && !is_object($object)) || (is_object($object) && $object->getId())) {
```

Comment:
I don't understand the purpose of this check.
The `$object` must be an object, otherwise the call to `resolveUriOption` would fail.
Moreover, we can't assume that a generic object has a `getId` method.

Reply:
More context:

- I have the form to create a new entity `Article`;
- `filePath` is property of related entity `private ?File $sourceFile = null;` and `#[ORM\ManyToOne(cascade: ['persist'])]` is set in `Article` for this relation;
- in `#[Vich\UploadableField` `fileNameProperty: 'sourceFile.filePath'` is set to inject data to `File` entity;
- VichImageType throws exception `PropertyAccessor requires a graph of objects or arrays to operate on, but it found type "NULL" while trying to traverse path "sourceFile.filePath" at property "filePath"`;
- `File` entity is optional in `Article` entity so I don't want to initialize it by `new File()` because cascade persist is used;
- VichImageType executes `resolveUriOption` on object which doesn't exist yet, so it doesn't make sens;
-  `$form->getParent()->getData()` returns `Article` not `File`;

Using `VichFileType` I can prevent this error by `'download_uri' => false` in form for new entity. There is no condition `if ($options['download_uri'] && $object) {` in `VichImageType` .

I can prepare next PR to add this missing condition to VichImageType, but if you expects here `$form->getParent()->getData()` an entity that has property for filePath, it dosen't work if related entity is used.

Reply:
So define your type class and use that instead of VichImageType.

Reply:
Better idea is to update VichImageType to support  'download_uri' => false in the same way like in VicuFileType. Now even if is false, it calls resolveUriOption.